jenkins_pipelines/manager_testsuite/Jenkinsfile_qa_testsuite

```diff
@@ -41,26 +41,32 @@ pipeline {
                     } else {
                       sh "bash jenkins-deploy.sh ${params.sumaform_env} ${params.testsuite_branch}"
                     }
-                    deployed = true 
+                    sh "bash create-snapshots.sh create_snapshots deploy"
+                    deployed = true
                 }
             }
         }
 
         stage('Core - Setup') {
             steps {
                 script { env.deployed = true }
+                sh "bash create-snapshots.sh revert_snapshots deploy"
```

Comment:
We need to are more logic on reverts, revert always is a waste of time, we need to revert only when : 
- The stage is re-started
- The snapshot exist

Reply:
I added a counter and conditionals to check if it restarted. 
If the snapshot doesn't exist it means it failed to create it before at the stage and the pipeline stopped there. We won't be able to rerun the next stage at all if the previous failed